body-tracking-samples/simple_3d_viewer/main.cpp

```diff
@@ -99,24 +104,195 @@ bool ParseInputSettingsFromArg(int argc, char** argv, InputSettings& inputSettin
+            }
+        }
+
+        // Visualize bones
+        for (size_t boneIdx = 0; boneIdx < g_boneList.size(); boneIdx++)
+        {
+            k4abt_joint_id_t joint1 = g_boneList[boneIdx].first;
+            k4abt_joint_id_t joint2 = g_boneList[boneIdx].second;
+
+            if (body.skeleton.joints[joint1].confidence_level >= K4ABT_JOINT_CONFIDENCE_LOW &&
+                body.skeleton.joints[joint2].confidence_level >= K4ABT_JOINT_CONFIDENCE_LOW)
+            {
+                bool confidentBone = body.skeleton.joints[joint1].confidence_level >= K4ABT_JOINT_CONFIDENCE_MEDIUM &&
+                    body.skeleton.joints[joint2].confidence_level >= K4ABT_JOINT_CONFIDENCE_MEDIUM;
+                const k4a_float3_t& joint1Position = body.skeleton.joints[joint1].position;
+                const k4a_float3_t& joint2Position = body.skeleton.joints[joint2].position;
+
+                window3d.AddBone(joint1Position, joint2Position, confidentBone ? color : lowConfidenceColor);
+            }
+        }
+    }
+
+    k4a_capture_release(originalCapture);
+    k4a_image_release(depthImage);
+
+}
+
+void PlayFile(InputSettings inputSettings) {
+    // Initialize the 3d window controller
+    Window3dWrapper window3d;
+
+    //create the tracker and playback handle
+    k4a_calibration_t sensor_calibration;
+    k4abt_tracker_t tracker = NULL;
+    k4a_playback_t playback_handle = NULL;
+
+    const char* file = inputSettings.FileName.c_str();
+    if (k4a_playback_open(file, &playback_handle) != K4A_RESULT_SUCCEEDED)
+    {
+        printf("Failed to open recording: %s\n", file);
+        return;
+    }
+
+
+    if (k4a_playback_get_calibration(playback_handle, &sensor_calibration) != K4A_RESULT_SUCCEEDED)
+    {
+        printf("Failed to get calibration\n");
+        return;
+    }
+    
+
+    k4a_capture_t capture = NULL;
+    k4a_stream_result_t result = K4A_STREAM_RESULT_SUCCEEDED;
+
+    k4abt_tracker_configuration_t tracker_config = { K4ABT_SENSOR_ORIENTATION_DEFAULT };
+
+    tracker_config.processing_mode = inputSettings.CpuOnlyMode ? K4ABT_TRACKER_PROCESSING_MODE_CPU : K4ABT_TRACKER_PROCESSING_MODE_GPU;
+
+    VERIFY(k4abt_tracker_create(&sensor_calibration, tracker_config, &tracker), "Body tracker initialization failed!");
+
+    k4abt_tracker_set_temporal_smoothing(tracker, 1);
 
+    int depthWidth = sensor_calibration.depth_camera_calibration.resolution_width;
+    int depthHeight = sensor_calibration.depth_camera_calibration.resolution_height;
+
+    window3d.Create("3D Visualization", sensor_calibration);
+    window3d.SetCloseCallback(CloseCallback);
+    window3d.SetKeyCallback(ProcessKey);
+
+    while (result == K4A_STREAM_RESULT_SUCCEEDED)
+    {
+        result = k4a_playback_get_next_capture(playback_handle, &capture);
+        if (result == K4A_STREAM_RESULT_SUCCEEDED)
+        {
+            
+            //enque capture and pop results - asynchronous
```

Comment:
If you are using K4A_WAIT_INFINITE, you are processing it synchronously (if you are directly copying from the offline processor, then it is wrong there as well...)

Reply:
What value should we be using instead? Or should we just update the comment to specify synchronous operation? I wrote the majority of the PlayFile code last summer and based it on the quickstart at the time, so that might explain if some of the variable names have changed since. It looks like offline_processor is also using K4A_WAIT_INFINITE

Reply:
Setting both function timeout K4A_WAIT_INFINITE is reasonable for playback. Updating the comment would be enough. 
Just FYI, in order to achieve the maximum performance, the best method in playback will be have two threads. One thread keeps feeding new frames using API k4abt_tracker_enqueue_capture with timeout K4A_WAIT_INFINITE. The other thread keeps poping frames using API k4abt_tracker_pop_result with timeout 0. But it will be advanced usage and maybe not worth putting in this simple example.